core/src/main/scala/org/apache/spark/sql/delta/commands/DeleteWithDeletionVectorsHelper.scala

```diff
@@ -313,7 +319,9 @@ object DeletionVectorBitmapGenerator {
       val filePathToDV = candidateFiles.map { add =>
         val serializedDV = Option(add.deletionVector).map(dvd => JsonUtils.toJson(dvd))
         // Paths in the metadata column are canonicalized. Thus we must canonicalize the DV path.
-        FileToDvDescriptor(absolutePath(basePath, add.path).toUri.toString, serializedDV)
+        FileToDvDescriptor(
+          SparkPath.fromPath(absolutePath(basePath, add.path)).urlEncoded,
```

Comment:
This was not changed in the 3.4 commit https://github.com/delta-io/delta/commit/5c3f4d37951ab4edf1cc3364ec6e8259844b331c#diff-ed9e118950c05851496615b00bb3ae1a2dc6879b8b0ab02c5b956026ee911ef4

Is it really required?

Reply:
It's not required. But using `SparkPath.urlEncoded` in the place of `Path.toUri.toString` is the right way to go forward.